pkg/util/proxyconfig/no_proxy.go

```diff
@@ -54,6 +54,7 @@ func mergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructur
 		"localhost",
 		".svc",
 		".cluster.local",
+		".hypershift.local",
```

Comment:
Since this is HyperShift specific it might be better to add it conditionally:
```
if hcpCfg := NewHyperShiftConfig(); hcpCfg.Enabled {
    set.Insert(".hypershift.local")
}
```